rcl_yaml_param_parser/src/parser.c

```diff
@@ -1278,6 +1659,201 @@ static rcutils_ret_t parse_value(
   return ret;
 }
 
+static rcutils_ret_t parse_descriptor(
+  namespace_tracker_t * ns_tracker,
+  const yaml_event_t event,
+  const bool is_seq,
+  const size_t node_idx,
+  const size_t parameter_idx,
+  rcl_params_t * params_st)
+{
+  RCUTILS_CHECK_ARGUMENT_FOR_NULL(params_st, RCUTILS_RET_INVALID_ARGUMENT);
```

Comment:
@bpwilcox consider adding a couple `assert`s to ensure both `*_idx` arguments are within bounds.